File: Python/2000.py
# 2000. Reverse Prefix of Word
# https://leetcode.com/problems/reverse-prefix-of-word/

class Solution:
  def reversePrefix(self, word: str, ch: str) -> str:
    first_ind = word.find(ch) + 1
    return word[0:first_ind][::-1] + word[first_ind:]

File: Rust/2000.rs
/**
 * 2000. Reverse Prefix of Word
 * https://leetcode.com/problems/reverse-prefix-of-word/
 */

impl Solution {
  pub fn reverse_prefix(word: String, ch: char) -> String {
    match word.find(ch) {
      None => {
        return word;
      }

      Some(first_ind) => {
        return [
          &word[0..first_ind + 1].chars().rev().collect::<String>(),
          &word[first_ind + 1..],
        ].concat();
      }
    }
  }
}
